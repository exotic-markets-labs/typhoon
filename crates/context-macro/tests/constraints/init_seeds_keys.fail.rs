use {
    bytemuck::{AnyBitPattern, NoUninit},
    pinocchio::pubkey::Pubkey,
    pinocchio_pubkey::declare_id,
    typhoon_account_macro::*,
    typhoon_accounts::*,
    typhoon_context_macro::*,
    typhoon_program_id_macro::program_id,
};

program_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");

#[derive(NoUninit, AnyBitPattern, AccountState, Copy, Clone)]
#[repr(C)]
pub struct Counter {
    pub count: u64,
}

#[context]
#[args(admin: Pubkey, bump: u8)]
pub struct InitContext {
    pub payer: Mut<Signer>,
    #[constraint(
        init,
        payer = payer,
        space = Counter::SPACE,
        seeds = [&args.admin],
        seeded = [&args.admin],
        bump
    )]
    pub counter: Mut<Account<Counter>>,
    pub system: Program<System>,
}

pub fn main() {}
