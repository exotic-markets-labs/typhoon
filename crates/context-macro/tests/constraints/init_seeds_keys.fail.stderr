error: Cannot specified keys and seeds at the same time.
  --> tests/constraints/init_seeds_keys.fail.rs:30:9
   |
30 |     pub counter: Mut<Account<Counter>>,
   |         ^^^^^^^
